temoa/data_io/loader_manifest.py

```
@@ -9,10 +9,17 @@
 
+from __future__ import annotations
+
 from dataclasses import dataclass, field
-from typing import Any, Optional
+from typing import TYPE_CHECKING, Any, TypeAlias
+
+if TYPE_CHECKING:
+    from pyomo.core import Param, Set
 
-from pyomo.core import Param, Set
+    ComponentType: TypeAlias = Set | Param
```

Ruff (0.14.0)

Type alias `ComponentType` uses `TypeAlias` annotation instead of the `type` keyword

Use the `type` keyword

(UP040)